test/celeritas/ext/ImportData.test.cc

```diff
@@ -0,0 +1,768 @@
+        EXPECT_EQ(ImportUnits::mev_per_cm, dedx.y_units);
+        ASSERT_EQ(2, dedx.physics_vectors.size());
+
+        const ImportPhysicsVector& steel = dedx.physics_vectors.back();
+        EXPECT_EQ(ImportPhysicsVectorType::log, steel.vector_type);
+        ASSERT_EQ(steel.x.size(), steel.y.size());
+        ASSERT_EQ(85, steel.x.size());
+        EXPECT_SOFT_EQ(1e-4, steel.x.front());
+        EXPECT_SOFT_EQ(1e8, steel.x.back());
+        EXPECT_SOFT_EQ(839.66835335480653, steel.y.front());
+        EXPECT_SOFT_EQ(11.380485677836065, steel.y.back());
+    }
+    {
+        // Test range table
+        const ImportPhysicsTable& range = tables[1];
+        ASSERT_EQ(ImportTableType::range, range.table_type);
+        EXPECT_EQ(ImportUnits::mev, range.x_units);
+        EXPECT_EQ(ImportUnits::cm, range.y_units);
+        ASSERT_EQ(2, range.physics_vectors.size());
+
+        const ImportPhysicsVector& steel = range.physics_vectors.back();
+        EXPECT_EQ(ImportPhysicsVectorType::log, steel.vector_type);
+        ASSERT_EQ(steel.x.size(), steel.y.size());
+        ASSERT_EQ(85, steel.x.size());
+        EXPECT_SOFT_EQ(1e-4, steel.x.front());
+        EXPECT_SOFT_EQ(1e8, steel.x.back());
+        EXPECT_SOFT_EQ(2.3818927937550707e-07, steel.y.front());
+        EXPECT_SOFT_EQ(8786971.3079055995, steel.y.back());
+    }
+    {
+        // Test cross-section table
+        const ImportPhysicsTable& xs = tables[2];
+        ASSERT_EQ(ImportTableType::lambda, xs.table_type);
+        EXPECT_EQ(ImportUnits::mev, xs.x_units);
+        EXPECT_EQ(ImportUnits::cm_inv, xs.y_units);
+        ASSERT_EQ(2, xs.physics_vectors.size());
+
+        const ImportPhysicsVector& steel = xs.physics_vectors.back();
+        EXPECT_EQ(ImportPhysicsVectorType::log, steel.vector_type);
+        ASSERT_EQ(steel.x.size(), steel.y.size());
+        ASSERT_EQ(54, steel.x.size());
+        EXPECT_SOFT_EQ(2.6269057995911775, steel.x.front());
+        EXPECT_SOFT_EQ(1e8, steel.x.back());
+        EXPECT_SOFT_EQ(0, steel.y.front());
+        EXPECT_SOFT_EQ(0.18987862452122845, steel.y[1]);
+        EXPECT_SOFT_EQ(0.43566778103861714, steel.y.back());
+    }
+    {
+        // Test model microscopic cross sections
+        auto get_values = [](const ImportProcess::ElementPhysicsVectorMap& xs) {
+            std::unordered_map<std::string, std::vector<double>> result;
+            for (const auto& kv : xs)
+            {
+                const auto& vec = kv.second;
+                result["x_size"].push_back(vec.x.size());
+                result["y_size"].push_back(vec.y.size());
+                result["x_front"].push_back(vec.x.front());
+                result["y_front"].push_back(vec.y.front() / units::barn);
+                result["x_back"].push_back(vec.x.back());
+                result["y_back"].push_back(vec.y.back() / units::barn);
+                result["element_id"].push_back(kv.first);
+            }
+            return result;
+        };
```

Comment:
A map isn't the right choice here because you know exactly what keys you have, and more importantly because the data types are heterogeneous (sizes should be integers, values should be reals). Make a struct, and you can define a `void print_expected() const` to have it print out the expected values to make updating easier: see e.g. `GeantImporterTest::ImportSummary::print_expected` .
```
struct ProcessXs
{
  std::vector<int> size;
  std::vector<double> x_front;
  std::vector<double> y_front;
  ...
};
```

Also, the check on `element_id` makes me realize that `ImportProcess` shouldn't have a `ElementPhysicsVectorMap`: it should just be a vector, one per element ID. All processes are going to have all elements, right?

Reply:
that's my bad :)

Reply:
@amandalund you're supposed to blame the reviewer for not catching it ;)

Reply:
I guess everything is @sethrj's fault

Reply:
> ElementPhysicsVectorMap: it should just be a vector, one per element ID. All processes are going to have all elements, right?

Okay, I remembered why we went for a `map`: we have one `map<element_id, physics_vector>` per material, and each material has a different set of `element_ids`. What spans over all entities is the `ModelMicroXS`, which spans over all materials for every model.

We could go for a
```cpp
struct
{
    int element_id;
    ImportPhysicsVector physics_vector;
};
```
and store a vector of structs instead of a map... but then we lose the handy `find(element_id)` function call.

Reply:
Oh duh, that's right. For this simple case they all have all elements in the whole problem. But, if we can guarantee the ordering of the element *components*, then we could make it a vector... at the cost of having to "reassemble" that metadata from the material compositions. What do you think?

Reply:
Not sure if I understood you correctly. Even if the ordering of the `ImportMatElemComponent` is right (i.e. `import_material.elements.at(3)` is the element component with `element_id == 3`), a given material could have `element_ids = {7, 13, 25}`, so in a vector we won't recover the correct ids by doing something like `element_phys_vectors.at(7)` has the `physics_vector` for `element_id == 7`.

Reply:
What I'm saying is if we make the mapping `ElementComponentId` rather than `ElementId` then an array is the right data structure. The `ElementComponentId` corresponds to `ElementId` via the material definition: it should be the sorted `ElementId` indices.